lib/util/source-code.js

```diff
@@ -88,68 +88,66 @@ function looksLikeExport(astNode) {
  * @param {ASTNode} ast - The Program node of the AST representing the code. This AST should be created from the text that BOM was stripped.
  * @constructor
  */
-function SourceCode(text, ast) {
-    validate(ast);
+class SourceCode {
```

Comment:
SourceCode is also exposed directly so people can pass in pre-computed ASTs (using new SourceCode()). So the larger concern is people calling SourceCode() or SourceCode.apply(), which are more likely than the other examples mentioned.